cpp/src/arrow/acero/exec_plan.cc

```diff
@@ -1001,18 +1000,24 @@ struct BatchConverter {
 
 Result<AsyncGenerator<std::shared_ptr<RecordBatch>>> DeclarationToRecordBatchGenerator(
     Declaration declaration, QueryOptions options,
-    ::arrow::internal::Executor* cpu_executor, std::shared_ptr<Schema>* out_schema) {
+    ::arrow::internal::Executor* cpu_executor, std::shared_ptr<Schema>* out_schema,
+    std::shared_ptr<ExecPlan>* out_plan) {
```

Comment:
Please document this argument

Reply:
What is the policy for documenting arguments? Most arguments are not documented, and in many cases their meaning is clear. I think this is the case here too.

Reply:
I think it is worthwhile to at least explain that the `out_plan` is used for cancellation, otherwise this API looks confusion - it is not intuitive to return exec plan in a method called DeclarationToRecordBatchGenerator)?